packages/react/src/components/table/Table.tsx

```diff
@@ -138,6 +138,13 @@ export type TableProps = React.ComponentPropsWithoutRef<'table'> & {
    * Sorting order applied for initial sorting.
    */
   initialSortingOrder?: 'asc' | 'desc';
+  /**
+   * Function for custom sorting or providing side effects before or after the sort. If you want to use the component's own sorting function, call the handleSort callback function.
+   * @param order
+   * @param colKey
+   * @param handleSort
+   */
+  onSort?: (order: 'asc' | 'desc', colKey: string, handleSort: () => void) => void;
```

Comment:
Remember to add this new prop to documentation site code tab props.

Reply:
Done. I put the same description as in the component prop doc.